### PR TITLE
Proposed new flow/structure

### PR DESCRIPTION
**Pull request by ytham**
- fixes typescript compile and package issues
- `app/axiom/data/inputs.json` is default input (from new sdk-client CLI that checks if that file exists`
- circuit inputs use inputs.xxx variables

### CODE REVIEW

File: app/axiom/average.circuit.ts [hunk 1]
```diff
@@ -34,21 +25,21 @@ export const circuit = async ({
...
 
   // Create an array of block numbers to sample from
   const blockNumbers = Array.from(
     {length: samples}, 
-    (_: any, i: number) => blockNumber.number() - (spacing * i)
+    (_: any, i: number) => inputs.blockNumber.number() - (spacing * i)
   );
 
   // Get all balances for the given address at the block numbers we are sampling from
   const balances = await Promise.all(
```
Comment (yi-sun) [inline app/axiom/average.circuit.ts]: We should avoid Promise.all since the order is not deterministic

File: lib/axiom-v2-contracts [hunk 2]
```diff
@@ -0,0 +1 @@
+Subproject commit 52245e9a5445fc88258a54168728fb3ab4460561
```
Comment (yi-sun) [inline lib/axiom-v2-contracts]: Should remove this dependency entirely to only use axiom-v2-client

File: package.json [hunk 3]
```diff
@@ -10,7 +10,7 @@
   "author": "Intrinsic Technologies",
   "license": "ISC",
   "dependencies": {
-    "@axiom-crypto/client": "0.2.2-rc2.0",
+    "@axiom-crypto/client": "link:../axiom-sdk-client/client/dist",
```
Comment (yi-sun) [inline package.json]: Remove this local patch

**Review by yi-sun — CHANGES_REQUESTED** [review]

**Comment by ytham** [comment]
@yi-sun I think the CI test is erroring due to the version check being only for one specific version: https://github.com/axiom-crypto/axiom-v2-client/blob/ff030872d7200cace1c9cb7ad5a028c1ade8a067/src/test/AxiomVm.sol#L274

we might want to consider just doing a loose version check or just a package existence check otherwise we might run into some issues down the line

**Review by yi-sun — APPROVED** [review]
LGTM

**Pull request merged**